build/docker/Dockerfile.release

```diff
@@ -1,53 +0,0 @@
-FROM alpine:latest
```

Comment:
Do you expect to replace this Dockerfile?  I was wondering how we will make official builds if the docker image isn't checking sha sums.  Thoughts?

Reply:
@jasonodonnell i've sent you some more info on the new private release repo, where the Release.dockerfile has been moved to. We could totally still have a copy of it here too if you'd like.

Reply:
Let's keep it here in case community members wants to build it from official binaries and shasums!

Reply:
Thanks, I have removed it from the release repo and put it back here, along with a new make target (prod-image) so we can build the production image from this repo. This target will be executed from the release pipeline (in the private release repo).